src/Core/src/Handlers/WebView/WebViewHandler.Windows.cs

```diff
@@ -279,20 +273,29 @@ void InitialCookiePreloadIfNecessary(string url)
 
 			if (cookies is not null)
 			{
-				var existingCookies = GetCookiesFromPlatformStore(url);
-				foreach (HttpCookie cookie in existingCookies)
+				var existingCookies = await GetCookiesFromPlatformStore(url);
+
+				if (existingCookies.Count == 0)
+					return;
+
+				foreach (CoreWebView2Cookie cookie in existingCookies)
 				{
+					// TODO Ideally we use cookie.ToSystemNetCookie() here, but it's not available for some reason check bac later
```

Comment:
```suggestion
					// TODO Ideally we use cookie.ToSystemNetCookie() here, but it's not available for some reason check back later
```
Stupid nit (I am the king of typos and won't let you dethrone me!)